include/swift/AST/SILOptions.h

```diff
@@ -121,7 +122,9 @@ class SILOptions {
   bool AssumeUnqualifiedOwnershipWhenParsing = false;
 
   /// Assume that code will be executed in a single-threaded environment.
-  bool AssumeSingleThreaded = false;
+  /// TODO: return to false when we pass "-Xfrontend -assume-single-threaded"
+  ///       to swift compiler while building standard library.
+  bool AssumeSingleThreaded = std::getenv("SWIFT_ASSUME_SINGLE_THREADED");
```

Comment:
Removed std::getenv in https://github.com/apple/swift/pull/7557/commits/8f8fd35237c1292c4d30bead54f120f6d92825f2. We can address the issue in a separate PR as needed.